src/app/components/todo-form/todo-form.component.html

```diff
@@ -1,12 +1,51 @@
-<div class="todo-form">
-  <div class="form-content">
-    <input #inputTask class="task-input" type="search" />
-    <!-- <input class="task-input-btn" type="button" value="Add Task"/> -->
-    <button
-      class="task-input-btn"
-      (click)="addNewTask(inputTask.value); ($event.preventDefault)"
-    >
-      Add Task
-    </button>
+<form [formGroup]="addTaskForm">
+  <div class="todo-form">
+    <div class="form-content">
+      <input
+        formControlName="task"
+        #inputTask
+        class="task-input"
+        type="search"
+      />
+
+      <button
+        class="task-input-btn"
+        [disabled]="inputTask.value == 0"
+        (click)="addNewTask(inputTask.value)"
+        type="submit"
+      >
+        Add Task
+      </button>
+    </div>
   </div>
+</form>
+
+<div
+  class="handleErrorMessage"
+  *ngIf="
+    addTaskForm.controls.task.errors &&
+    addTaskForm.controls.task.touched &&
+    addTaskForm.controls.task.dirty
+  "
+>
+  <p class="errorMessage" *ngIf="addTaskForm.controls.task.errors.required">
+    ** You Must write Task to Add it .
+  </p>
+  <p class="errorMessage" *ngIf="addTaskForm.controls.task.errors.minlength">
+    ** Task must be greater than
+    {{ addTaskForm.controls.task.errors.minlength.actualLength }} Char but Task
+    should be at least
+    {{ addTaskForm.controls.task.errors.minlength.requiredLength }} Char
+  </p>
+  <p class="errorMessage" *ngIf="addTaskForm.controls.task.errors.maxlength">
+    ** Task must be less than
+    {{ addTaskForm.controls.task.errors.maxlength.actualLength }} Char but Task
+    should be at more
+    {{ addTaskForm.controls.task.errors.maxlength.requiredLength }} Char
+  </p>
+  <p class="errorMessage" *ngIf="addTaskForm.controls.task.errors.pattern">
+    ** Tasks name should be strings Only <br />
+    - Numbers not accepted <br />
```

Comment:
It will accepts numbers with strings, but numbers only not accepted also numbers and special characters without actual words not accepted.

Reply:
cant find suitable regex for this

Reply:
Let's park it now, we will do it later.
No problem!